sql/core/src/test/scala/org/apache/spark/sql/DynamicPartitionPruningSuite.scala

```diff
@@ -1342,6 +1345,52 @@ abstract class DynamicPartitionPruningSuiteBase
       }
     }
   }
+
+  test("SPARK-32748: propagate local properties to dynamic pruning thread") {
+    def checkPropertyValueByUdfResult(propKey: String, propValue: String): Unit = {
+      spark.sparkContext.setLocalProperty(propKey, propValue)
+      val df = sql(
+        s"""
+           |SELECT compare_property_value(f.date_id, '$propKey', '$propValue') as col
+           |FROM fact_sk f
+           |INNER JOIN dim_store s
+           |ON f.store_id = s.store_id AND s.country = 'NL'
+          """.stripMargin)
+
+      checkPartitionPruningPredicate(df, false, true)
+      assert(df.collect().forall(_.toSeq == Seq(true)))
+    }
+
+    try {
+      SQLConf.get.setConf(StaticSQLConf.BROADCAST_EXCHANGE_MAX_THREAD_THRESHOLD, 1)
```

Comment:
does this really work? It's a static conf. As long as there is an earlier test triggers DPP, this won't work.

Reply:
Yes you are right.. but it's not because it's a static conf, it's because `executionContext` is in the `SubqueryBroadcastExec` object.
This makes it hard to write unit test. Do you have any suggestion?

Reply:
Actually it was the same before: https://github.com/apache/spark/pull/27266/files#diff-6dabaf3c491be73cc0479b1f368d157bR164

Can we move this test to there? Then it's easier to read as these special tests are put together.

Reply:
Thanks for the advice! Updated.